### PR TITLE
Move kopia storage consts to a secrets package

### PR DESCRIPTION
**Pull request by kale-amruta**
## Change Overview

Move the kopia location secret consts to `secrets` package so that they can be reused. We cannot use the consts from storage package in secrets since it creates cyclic dependency. Hence moving it to secrets package. This PR would be the base PR for validation of kopia repository server secrets PR - https://github.com/kanisterio/kanister/pull/1940

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/secrets/const.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+type LocType string
+
+const (
+	// Location types
+	LocTypeS3        LocType = "s3"
+	LocTypeGCS       LocType = "gcs"
+	LocTypeAzure     LocType = "azure"
+	LocTypeFilestore LocType = "filestore"
+)
```
[the remark this answers is not in the document]
Comment (kale-amruta) [reply pkg/secrets/const.go:11]: I did not understand. I actually wanted to reuse these `storage` constants in  `secrets` package. But `storage` is already importing `secrets` package. Hence it causes cyclic dependency issue if I try to import it again. This change is in response to this comment - https://github.com/kanisterio/kanister/pull/1940#discussion_r1176451951`

File: pkg/kopia/command/storage/azure_args_test.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ package storage
 import (
 	"fmt"
 
+	"github.com/kanisterio/kanister/pkg/secrets/repositoryserver"
```
Comment (PrasadG193) [inline pkg/kopia/command/storage/azure_args_test.go:20]: Let's keep the Kanister imports in a separate group

File: pkg/kopia/command/storage/filesystem_args_test.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ package storage
 import (
 	"fmt"
 
+	"github.com/kanisterio/kanister/pkg/secrets/repositoryserver"
```
Comment (PrasadG193) [inline pkg/kopia/command/storage/filesystem_args_test.go:20]: same here

File: pkg/kopia/command/storage/s3_args_test.go [hunk 4]
```diff
@@ -17,6 +17,7 @@ package storage
 import (
 	"fmt"
 
+	"github.com/kanisterio/kanister/pkg/secrets/repositoryserver"
```
Comment (PrasadG193) [inline pkg/kopia/command/storage/s3_args_test.go:20]: same here

File: pkg/kopia/command/storage/storage_args_test.go [hunk 5]
```diff
@@ -17,6 +17,7 @@ package storage
 import (
 	"fmt"
 
+	"github.com/kanisterio/kanister/pkg/secrets/repositoryserver"
```
Comment (PrasadG193) [inline pkg/kopia/command/storage/storage_args_test.go:20]:
```suggestion
	"github.com/kanisterio/kanister/pkg/secrets/repositoryserver"
	
```

File: pkg/secrets/repositoryserver/const.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	LocTypeGCS       LocType = "gcs"
+	LocTypeAzure     LocType = "azure"
+	LocTypeFilestore LocType = "filestore"
+
+	// LocationSecretType represents the storage location secret type for kopia repository server
+	Location corev1.SecretType = "secrets.kanister.io/storage-location"
+	// RepositoryPasswordSecretType represents the kopia repository passowrd secret type
+	RepositoryPassword corev1.SecretType = "secrets.kanister.io/kopia-repository/password"
+	// RepositoryServerAdminCredentialsSecretType represents the kopia server admin credentials secret type
+	RepositoryServerAdminCredentials corev1.SecretType = "secrets.kanister.io/kopia-repository/serveradmin"
```
Comment (PrasadG193) [inline pkg/secrets/repositoryserver/const.go:34]: Mismatch between const name used in comments and actual const.
e.g Comments is for `LocationSecretType` but actual const is `Location`

**Review by ankitjain235 — APPROVED** [review]

**Pull request merged**